#! /bin/sh -e

# DP: Fix compatibility of old and new nice interfaces. Patch by HJ Lu

if [ $# -ne 2 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch) patch -d "$2" -f --no-backup-if-mismatch -p1 < $0;;
    -unpatch) patch -d "$2" -f --no-backup-if-mismatch -R -p1 < $0;;
    *)
	echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
	exit 1
esac
exit 0

--- glibc-2.2.5/resource/Versions.nice	Tue Jan  4 08:11:53 2000
+++ glibc-2.2.5/resource/Versions	Wed Apr 10 10:20:35 2002
@@ -22,4 +22,8 @@ libc {
     # s*
     setrlimit64;
   }
+  GLIBC_2.2.6 {
+    # The new nice interface
+    nice;
+  }
 }
--- glibc-2.2.5/sysdeps/unix/sysv/linux/nice.c.nice	Wed Mar 13 04:52:29 2002
+++ glibc-2.2.5/sysdeps/unix/sysv/linux/nice.c	Wed Apr 10 10:31:37 2002
@@ -1 +1,5 @@
+#define nice __new_nice
 #include <sysdeps/unix/nice.c>
+#undef nice
+#include <shlib-compat.h>
+versioned_symbol (libc, __new_nice, nice, GLIBC_2_2_5);
--- glibc-2.2.5/sysdeps/unix/sysv/linux/syscalls.list.nice	Fri Apr 20 13:49:19 2001
+++ glibc-2.2.5/sysdeps/unix/sysv/linux/syscalls.list	Wed Apr 10 10:04:03 2002
@@ -31,6 +31,7 @@ mremap		EXTRA	mremap		b:aini	__mremap	mr
 munlock		-	munlock		i:ai	munlock
 munlockall	-	munlockall	i:	munlockall
 nanosleep	-	nanosleep	i:pp	__libc_nanosleep	__nanosleep nanosleep
+oldnice		EXTRA	nice		i:i	__old_nice	nice@GLIBC_2.0
 nfsservctl	EXTRA	nfsservctl	i:ipp	nfsservctl
 pause		-	pause		i:	__libc_pause	pause
 personality	init-first personality	i:i	__personality	personality
